### DISCUSSION TITLE
Required fields in subschema

### DISCUSSION THREAD

**Discussion by AdamOndrejkovic** [json-schema-org/community#112] [Q&A]
Hi,

I have a functional schema and I've added a subschema that has required fields. Now I got an error as the object isn't always needed but the fields in subschema are. How can I achieve that the fields in subschema will be only required if there is an object of that type in the parent schema?

Thanks.

**Comment by karenetheridge** [comment]
See the if/then/else and dependentSchemas keywords.

https://json-schema.org/understanding-json-schema/reference/index.html

**Reply by AdamOndrejkovic** [reply]
Thank you I will check that I used dependentRequired and I got error of unknown keyword

**Reply by karenetheridge** [reply]
If you're using specification version draft7 (or earlier), then it's `dependencies`.
